libraries/protocol/htlc.cpp

```diff
@@ -38,8 +38,8 @@ namespace graphene { namespace protocol {
    {
       uint64_t days = ( claim_period_seconds + SECONDS_PER_DAY - 1 ) / SECONDS_PER_DAY;
       // multiply with overflow check
-      uint64_t per_day_fee = fee_params.fee_per_day * days;
-      uint64_t per_kb_fee = 0;
+      share_type per_day_fee = fee_params.fee_per_day * days;
```

Comment:
This doesn't avoid a potential overflow on `fee_params.fee_per_day * days`, although there is a check below.
Perhaps adopt the method used in `transfer_operation` which IMO is the best practice.
Actually there are same issues in other functions of this file as well as in other files e.g. `asset_ops.cpp`.